include/chopper/layout/layout.hpp

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include <iosfwd>
+#include <string>
+#include <vector>
+
+#include <chopper/prefixes.hpp>
+
+namespace chopper::layout
+{
+
+// Currently, the layout is structured by user bin.
+struct layout
+{
+    struct max_bin
+    {
+        std::vector<size_t> previous_TB_indices{}; // identifies the IBF based on upper levels
+        size_t id{};                               // the technical bin id that has the maximum kmer content
+
+        auto operator<=>(max_bin const &) const = default;
+
+        // needs a template (instead of using std::ostream directly) to be able to only include <iosfwd>
+        template <typename stream_type>
+            requires std::derived_from<stream_type, std::ostream>
+        friend stream_type & operator<<(stream_type & stream, max_bin const & object)
+        {
+            stream << prefix::header << prefix::merged_bin << "_";
+            auto it = object.previous_TB_indices.begin();
+            if (it != object.previous_TB_indices.end())
+                stream << *it++;
+            for (; it != object.previous_TB_indices.end(); ++it)
+                stream << ';' << *it;
+            stream << " max_bin_id:" << object.id;
+
+            return stream;
+        }
+    };
+
+    struct user_bin
+    {
+        std::string id{};                          // the id of the user bin. Often its filename.
+        std::vector<size_t> previous_TB_indices{}; // previous technical bin indices which refer to merged bin indices.
+        size_t number_of_technical_bins{};         // 1 == signle bin, >1 == split_bin
+        size_t storage_TB_id{}; // the id of the technical bin that the user bin is actullly stored in
+
+        auto operator<=>(user_bin const &) const = default;
+
+        // needs a template (instead of using std::ostream directly) to be able to only include <iosfwd>
+        template <typename stream_type>
```

Comment:
```suggestion
        template <typename stream_type = std::ostream>
```

Reply:
This causes an ambiguous overload error with the debug_stream_type...

<details>
```bash
/home/mi/smehringer/AGABI/Chopper/lib/seqan3/include/seqan3/core/debug_stream/range.hpp:100:27: error: static assertion failed: The reference type of the passed range cannot be streamed into the debug_stream.
  100 |     static_assert(detail::reference_type_is_streamable_v<rng_t, char_t>,
      |                   ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/mi/smehringer/AGABI/Chopper/lib/seqan3/include/seqan3/core/debug_stream/range.hpp:100:27: note: ‘seqan3::detail::reference_type_is_streamable_v<const std::vector<chopper::layout::layout::max_bin>&, char>’ evaluates to false
/home/mi/smehringer/AGABI/Chopper/lib/seqan3/include/seqan3/core/debug_stream/range.hpp:108:11: error: ambiguous overload for ‘operator<<’ (operand types are ‘seqan3::debug_stream_type<char>’ and ‘const chopper::layout::layout::max_bin’)
  108 |         s << *b;
      |         ~~^~~~~

```
</details>